inception-doc/src/main/resources/META-INF/asciidoc/developer-guide/release.adoc

```diff
@@ -0,0 +1,101 @@
+= Releasing INCEpTION
+
+== Prerequisites
+
+When deploying to an Artifactory instance, make sure that the UKP Maven Repository is with your
+user credentials are added to your machines `settings.xml`. In our case, the repository
+`ukp-oss-releases` has to be listed with a user and password that belongs to the `trusted-user`
```

Comment:
I would point to the Maven documentation on authentication: https://maven.apache.org/settings.html#Servers - in particular since that describes how developers can locally encrypt their credentials.

The details of the repo configuration (i.e. the user group) is something the repo admins need to know - the developers do not need to know it and I believe they cannot even see to which groups they belong.